File: 908. Smallest Range I/code.cpp
class Solution {
public:
    int smallestRangeI(vector<int>& A, int K) {
        int min_v=INT_MAX,max_v=INT_MIN;
        for(int i=0;i<A.size();i++)
        {
            if(min_v>A[i])
            {
                min_v=A[i];
            }
            if(max_v<A[i])
            {
                max_v=A[i];
            }
        }
        return max(0,max_v-min_v-2*K);
        
    }
};

File: 3. Longest Substring Without Repeating Characters/code.java
class Solution {
    public int max(int a,int b)
    {
        if(a>b)
        {
            return a;
        }
        return b;
    }
    public int lengthOfLongestSubstring(String s) {
        int size=s.length();
        int maxE=0;
        int a[]=new int[256];
        for(int i=0;i<=255;i++) a[i]=0;
        for(int i=0,j=0;j<size;j++)
        {
            int h=s.charAt(j);
            if(a[h]!=0)
            {
                i=max(a[h],i);
            }
            maxE=max(maxE,j-i+1);
            a[h]=j+1;
            // cout<<maxE<<endl;
        }
        
        return maxE;
    }
}
